client/nearby/src/components/workspace/Workspace.jsx

```diff
@@ -1,48 +1,41 @@
 /* eslint-disable react/prop-types */
 import React from 'react';
-import LoadingWorkspace from './LoadingWorkspace';
 
-const Workspace = ({ location: { workspaceId, neighborhood }, allInfo, pic }) => {
-    if (allInfo.length === 0) {
-        return <LoadingWorkspace />;
-    }
+function Amenities({ amenities }) {
+    const rest = amenities.length - 5;
 
-    // conditionally render array of amenities data
-    function Amenities({ amenities: { amenities } }) {
-        if (!amenities) {
-            return <></>;
-        }
-        const rest = amenities.length - 5;
-        return (
-            <>
-                {amenities.slice(0, 5).map((am) => (
-                    <li key={`${am.name}-${am.id}`}>
-                        &#8226;
-                        {am.name}
-                    </li>
-                ))}
-                <br />
-                {rest > 0 && <li> + {rest} more </li>}
-            </>
-        );
-    }
+    return (
+        <React.Fragment>
+            {amenities.slice(0, 5).map((am) => (
+                <li key={`${am.name}-${am._id}`}>
+                    {am.name + ' '}
+                    &#8226;{' '}
+                </li>
+            ))}
```

Comment:
If you don't need to copy `amenities`, `.slice` is inefficient. Faster would be this:
```
{amenities.map((am, i) => i < 5 ? ... : null)}
```

Reply:
Nice catch. I will fix this as well. Thanks Josh